incubator/orm/index.go

```diff
@@ -11,23 +11,23 @@ import (
 	"github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-// indexer creates and modifies the second index based on the operations and changes on the primary object.
+// indexer creates and modifies the second MultiKeyIndex based on the operations and changes on the primary object.
 type indexer interface {
 	OnCreate(store sdk.KVStore, rowId uint64, value interface{}) error
 	OnDelete(store sdk.KVStore, rowId uint64, value interface{}) error
 	OnUpdate(store sdk.KVStore, rowId uint64, newValue, oldValue interface{}) error
 }
 
-// index is the entry point for all index related operations.
-type index struct {
+// MultiKeyIndex
+type MultiKeyIndex struct {
```

Comment:
Better name, and nice to be public.
But the godoc on top needs some help :slightly_smiling_face: